### PR TITLE
Fix high and critical vulnerabilities

### PR DESCRIPTION
**Pull request by kararade**
**Release Type:** *Dev Improvement*

Fixes https://github.com/x-team/xp/issues/2046 and https://github.com/x-team/xp/issues/1974

## Description

It is fixing the critical vulnerabilities and most of the highs, the only high vulnerability left is related to the `parsejson` package that doesn't have a patch to fix it yet.

## Checklist

**Before submitting a pull request,** please make sure the following is done:

- [x] set yourself as an assignee
- [x] make sure your code lints (`npm run lint`)
- [x] Flow typechecks passed (`npm run flow`)
- [x] check cleanup tasks (https://github.com/x-team/xp/labels/cleanup) and take a suitable small one (if exists) in a related area of the current changes
- [ ] if added a new UI component that is meant to be used directly by other apps, remember to export that from the appropriate bundle file: `src/index.js` and/or `src/registration.js`
- [ ] component's documentation (`.stories.js` file) is changed or added accordingly to reflect any new or updated use cases and/or debug cases
- [ ] if you've fixed a bug, make sure to also include a new story that will expose declarations with problematic data that caused the bug in the first place, so that we can assure that no regressions will pop up on future interactions
- [x] if any snapshots have been changed, verify that component still works and looks as expected and update the changed snapshot (**NOTE:** This is temporarily disabled. Skip it.)
- [ ] once finished with all the changes, you run `npm run build` command and committed changed files to the branch
- [ ] **manually tested the app** by running it in several different browsers (Firefox, Chrome, Opera, Safari, MS IE/Edge, etc.) and checked nothing is broken and operates as expected!

## Steps to Test or Reproduce

In `xp-ui`:

1. Run `rm -rf node_modules/ && npm i`
2. Run `npm audit`
3. Observe that the report of vulnerabilities has none marked as critical and just 1 as high

## Screenshots

- Current status of the vulnerabilities

Screen Shot on 2020-04-10 at 02:10:57

- The only `parsejson` package that can't be fixed yet

Screen Shot 2020-03-25 at 02 03 45

- How is this high vulnerability affecting the project? 

The article published here said https://npmjs.com/advisories/528

> Affected versions of parsejson are vulnerable to a regular expression denial of service when parsing untrusted user input.

### CODE REVIEW

**Comment by nicksp** [comment]
@kararade Did you ensure apps works correctly after upgrading some libs like `storybook`-related ones? Have you run the app and went through the critical use paths to ensure this still operates as before? 

I'd also recommend you add these steps over to `Steps to Test or Reproduce` section as this is critical to verify and approve your changes. In other words, how do we know the app doesn't fail after upgrade?! Hope you got the points 😏

File: src/components/ui/ApplicantScreen.js [hunk 1]
```diff
@@ -21,11 +21,11 @@ const cx = {
 
   header: cmz(`
     & {
-      z-index: 2
       position: relative
       height: 50px
       display: flex
       flex-direction: column
+      z-index: 1000
```
Comment (nicksp) [inline src/components/ui/ApplicantScreen.js]: This change isn't part of your changelist. Please rebase with `develop` to get only relevant changes.

Comment (kararade) [reply src/components/ui/ApplicantScreen.js]: updated and removed

**Comment by kararade** [comment]
> Hey @kararade I don't see any actual changes on a PR. Can it be closed or would you like to work further on it?

I've fixed the last few conflicts encountered and tested the project. Everything looks good, although it's hard for me to confirm this since I don't know in-depth all the details of every element of the XP-UI.

However, I feel confident that the changes applied do not affect the project in a major way. I can say this because finally, the applied changes do not change in anything the versions of the current existing packages, but only update the `package-lock.json` as expected after running the `npm audit` command safely.

Let me know your thoughts

Thanks

**Pull request closed without merging**